src/use-visitor-data.ts

```diff
@@ -31,13 +30,13 @@ export function useVisitorData<TExtended extends boolean>(
 ): VisitorQueryContext<TExtended> {
   assertIsTruthy(getOptions, 'getOptions')
 
-  const previousGetOptions = usePrevious(getOptions)
-
   const { immediate } = config
   const { getVisitorData } = useContext<FpjsContextInterface<TExtended>>(FpjsContext)
 
-  const initialState = { isLoading: config.immediate ? true : false }
-  const [state, setState] = useState<QueryResult<VisitorData<TExtended>>>(initialState)
+  const [currentGetOptions, setCurrentGetOptions] = useState<UseVisitorDataOptions<TExtended>>(getOptions)
+  const [queryState, setQueryState] = useState<QueryResult<VisitorData<TExtended>>>({
+    isLoading: config.immediate,
```

Comment:
nit: I think we can just do this for consistency, since `immediate` is destructured from `config` above

```suggestion
    isLoading: immediate,
```